### PR TITLE
Use make URLs internal for load_audio and load_images

### LINKED ISSUE #2685: turicreate.load_audio broken with home directory present.
The following code generates a dataset that has no entries: 

```
In [13]: import turicreate as tc
    ...:
    ...: data = tc.load_audio("~/data/ESC-50/audio/")
```

However, the same code where the ~ is replaced by the absolute path name works fine.

### PR DESCRIPTION
**Pull request by TobyRoseman**
Fixes #2685

### CODE REVIEW

File: src/python/turicreate/toolkits/image_analysis/image_analysis.py [hunk 1]
```diff
@@ -62,7 +62,9 @@ def load_images(
     ...                                                       recursive=True)
     """
     from ... import extensions as _extensions
+    from ...util import _make_internal_url
```
Comment (guihao-liang) [inline src/python/turicreate/toolkits/image_analysis/image_analysis.py]: mixing relative and absolute is not a good idea in our project. We'd better choose one style and be coherent with it.

Comment (TobyRoseman) [reply src/python/turicreate/toolkits/image_analysis/image_analysis.py]: I haven't heard of that being a problem. What's your concern?

Comment (guihao-liang) [reply src/python/turicreate/toolkits/image_analysis/image_analysis.py]: > Relative imports for intra-package imports are highly discouraged. Always use the absolute package path for all imports. Even now that PEP 328 [7] is fully implemented in Python 2.5, its style of explicit relative imports is actively discouraged; absolute imports are more portable and usually more readable.

Comment (guihao-liang) [reply src/python/turicreate/toolkits/image_analysis/image_analysis.py]: or you can find it SO,
https://stackoverflow.com/questions/4209641/absolute-vs-explicit-relative-import-of-python-module

**Pull request merged**